packages/core/test/zodiac/OptimisticGovernor.js

```diff
@@ -1,691 +1,601 @@
+        event.proposal.transactions[1].value == 0 &&
+        event.proposal.transactions[1].data == txnData2 &&
+        event.proposal.transactions[1].operation == 0 &&
+        event.proposal.transactions[2].to == testToken2.options.address &&
+        event.proposal.transactions[2].value == 0 &&
+        event.proposal.transactions[2].data == txnData3 &&
+        event.proposal.transactions[2].operation == 0
+    );
+  });
+
+  it("Can not delete proposal that was not disputed", async function () {});
+
+  it("Disputed proposals can not be executed", async function () {
+    // Issue some test tokens to the avatar address.
+    await testToken.methods.allocateTo(avatar.options.address, toWei("3")).send({ from: accounts[0] });
+    await testToken2.methods.allocateTo(avatar.options.address, toWei("2")).send({ from: accounts[0] });
+
+    // Construct the transaction data to send the newly minted tokens to proposer and another address.
+    const txnData1 = constructTransferTransaction(proposer, toWei("1"));
+    const txnData2 = constructTransferTransaction(rando, toWei("2"));
+    const txnData3 = constructTransferTransaction(proposer, toWei("2"));
+    const operation = 0; // 0 for call, 1 for delegatecall
+
+    // Send the proposal with multiple transactions.
+    const transactions = [
+      { to: testToken.options.address, operation, value: 0, data: txnData1 },
+      { to: testToken.options.address, operation, value: 0, data: txnData2 },
+      { to: testToken2.options.address, operation, value: 0, data: txnData3 },
+    ];
+
+    const explanation = utf8ToHex("These transactions were approved by majority vote on Snapshot.");
+
+    const receipt = await optimisticOracleModule.methods
+      .proposeTransactions(transactions, explanation)
+      .send({ from: proposer });
+
+    const { ancillaryData } = (await findEvent(receipt, optimisticOracle, "ProposePrice")).match.returnValues;
+
+    const proposalTime = parseInt(await optimisticOracleModule.methods.getCurrentTime().call());
+
+    // Advance time to one second before end of the dispute period.
+    const stillOpen = liveness - 1;
+    await advanceTime(stillOpen);
+
+    // Dispute.
+    await optimisticOracle.methods
+      .disputePrice(optimisticOracleModule.options.address, identifier, proposalTime, ancillaryData)
+      .send({ from: disputer });
+
+    // Advance time past the liveness window.
+    await advanceTime(2);
+
+    // Proposal should not be executed.
+    assert(
+      await didContractThrow(optimisticOracleModule.methods.executeProposal(transactions).send({ from: executor }))
+    );
+  });
+
+  // /*
+  //  * This test is currently failing. Need to reason about how to send a transaction from the avatar
+  //  * to the module itself, with the transaction being proposed and approved through the module.
+  //  */
+  // it("Owner can delete proposals before execution", async function () {
```

Comment:
this would be a great test. What if you made the avatar address just an owner? Does the avatar have to be a Safe contract for the sake of this unit test?

Reply:
The Avatar should be the Safe address since we'll be sending transactions through the OG to the Safe to modify the OG module. I'll make this one a priority this week since it's important that it works before anyone can switch off their multi-sig. It's also something we can more easily test on testnet and mainnet, so we should plan to do that.